scripts/heatView.py

```diff
@@ -371,16 +397,18 @@ def plot_2D_map(ax, tmin, tmax, xdim, ydim, xwidth, ywidth, title_message, tempe
         #beginning of 3D plotting
         else:
             ax = fig.add_subplot(gs[0], projection='3d')
+            cache_in_z = 1 if cache_l3 and cache_l3_stacked else 0
                         #ind = yy + cores_in_y*xx + cores_in_x*cores_in_y*zz
 
             #core
             if (arch_type=="3D"):
-                xwidth=(float)(banks_in_x)/cores_in_x
+                cores_and_l3_in_x = cores_in_x + cache_l3_width
+                xwidth=(float)(banks_in_x)/cores_and_l3_in_x
```

Comment:
Do we assume that cache_l3_width is in x-dimension only? Is there any notion of cache height or y-dimension also?